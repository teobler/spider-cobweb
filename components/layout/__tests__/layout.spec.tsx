import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Aside from '../aside';
import Content from '../content';
import Layout from '../layout';

describe('<Layout />', () => {
  it('should render a empty layout', () => {
    const layoutJson = renderer.create(<Layout />).toJSON();
    expect(layoutJson).toMatchSnapshot();
  });

  it('detect the sider as children', () => {
    const wrapper = mount(
    <Layout>
      <Aside>Sider</Aside>
      <Content>Content</Content>
    </Layout>);
    expect(wrapper.find('.spider-layout').hasClass('spider-layout-with-aside')).toBe(true);
  });
});
