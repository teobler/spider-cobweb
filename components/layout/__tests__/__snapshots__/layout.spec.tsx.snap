// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render a empty layout 1`] = `
<section
  className="spider-layout"
/>
`;
